src/common/validators.ts

```diff
@@ -0,0 +1,243 @@
+ * @throws Error if any of the {@link domain} chunks is blank.
+ * @throws Error if any of the {@link domain} chunks is empty.
+ * @throws Error if any of the {@link domain} chunks has spaces.
+ */
+function validateDomain(domain: any, options?: ValidateDomainOptions): void {
+  // logger.debug('validating domain:', domain);
+
+  const { allowHierarchy = true } = options ?? {};
+
+  if (isEmpty(domain)) {
+    throw new Error(logger.createMessage(`Invalid domain. It cannot be empty.`));
+  }
+
+  if (typeof domain === 'boolean') {
+    throw new Error(logger.createMessage(`Invalid domain. It cannot be a boolean.`));
+  }
+
+  if (typeof domain === 'object') {
+    if (Array.isArray(domain)) {
+      throw new Error(logger.createMessage(`Invalid domain. It cannot be an array.`));
+    }
+    throw new Error(logger.createMessage(`Invalid domain. It cannot be an object.`));
+  }
+
+  domain = String(domain);
+
+  if (isContentfulString(domain)) {
+    if (!allowHierarchy && domain.includes('/')) {
+      throw new Error(logger.createMessage(`Invalid domain '${domain}'. It cannot have multiple chunks here.`));
+    }
+
+    const chunks = domain.split('/');
+
+    for (const chunk of chunks) {
+      if (chunk.includes(' ')) {
+        throw new Error(logger.createMessage(`Invalid domain '${domain}'. It cannot have empty spaces in '${chunk}'.`));
+      }
+
+      if (isEmptyOrBlankString(chunk)) {
+        throw new Error(logger.createMessage(`Invalid domain '${domain}'. It cannot have empty chunks.`));
+      }
+    }
+  } else {
+    throw new Error(logger.createMessage(`Invalid domain. It cannot be blank.`));
+  }
+}
+
+/**
+ * Validates name or identifier.
+ *
+ * @param nameOrIdentifier the name or identifier to validate.
+ *
+ * @throws Error if the {@link nameOrIdentifier} is blank.
+ * @throws Error if the {@link nameOrIdentifier} is empty.
+ * @throws Error if the {@link nameOrIdentifier} is a boolean.
+ * @throws Error if the {@link nameOrIdentifier} is an array.
+ * @throws Error if the {@link nameOrIdentifier} is an object.
+ * @throws Error if the {@link nameOrIdentifier} has chunks.
+ * @throws Error if the {@link nameOrIdentifier} has spaces.
+ */
+function validateNameOrIdentifier(nameOrIdentifier: any): void {
+  // logger.debug('validating name or identifier:', nameOrIdentifier);
+
+  if (isEmpty(nameOrIdentifier)) {
+    throw new Error(logger.createMessage(`Invalid name or identifier. It cannot be empty.`));
+  }
+
+  if (typeof nameOrIdentifier === 'boolean') {
+    throw new Error(logger.createMessage(`Invalid name or identifier. It cannot be a boolean.`));
+  }
+
+  if (typeof nameOrIdentifier === 'object') {
+    if (Array.isArray(nameOrIdentifier)) {
+      throw new Error(logger.createMessage(`Invalid name or identifier. It cannot be an array.`));
+    }
+    throw new Error(logger.createMessage(`Invalid name or identifier. It cannot be an object.`));
+  }
```

Comment:
Same as the domain comment, I think that we should enforce both domains and identifiers as strings.